pkg/operator/watcher/filewatcher.go

```diff
@@ -0,0 +1,111 @@
+package watcher
+
+import (
+	"fmt"
+	"io/ioutil"
+	"sync"
+
+	"gopkg.in/fsnotify.v1"
+
+	logf "github.com/openshift/cluster-ingress-operator/pkg/log"
+)
+
+var log = logf.Logger.WithName("filewatcher")
+
+// FileWatcher watches a file for changes.
+type FileWatcher struct {
+	sync.Mutex
+	fileName string
+	watcher  *fsnotify.Watcher
+}
+
+// New returns a new FileWatcher watching the given file.
+func New(file string) (*FileWatcher, error) {
+	fw := &FileWatcher{
+		fileName: file,
+	}
+
+	// Initial read of file.
+	if err := fw.ReadFile(); err != nil {
+		return nil, err
```

Comment:
Is the intent to make the operator container restart as long as the file is absent or empty?